### PR TITLE
feat: add filecoin chains

### PR DESCRIPTION
**Pull request by AmeanAsad**
## Motivation

Adding Filecoin chains into the `Chain` enum. Filecoin is launching its own virtual machine and adding the Filecoin chains into `ethers-rs` allows developers to use them in the rust web3 tooling. The main personal use I have is estimating chain gas costs when using `Foundry` to deploy smart contracts on the Filecoin virtual machine. 

## Solution

Added filecoin chains into the `Chain` enum and the other corresponding spots required for their definition. 

## PR Checklist

-   [ ] Added Tests (N/A) 
-   [ ] Added Documentation (N/A)
-   [ ] Updated the changelog 
-   [ ] Breaking changes

### CODE REVIEW

File: ethers-core/src/types/chain.rs [hunk 1]
```diff
@@ -299,6 +303,9 @@ impl Chain {
             Chiado => {
                 ("https://blockscout.chiadochain.net/api", "https://blockscout.chiadochain.net")
             }
+            FilecoinHyperspaceTestnet => {
+                ("https://api.hyperspace.node.glif.io/rpc/v1", "https://hyperspace.filfox.info")
+            }
```
Comment (mattsse) [inline ethers-core/src/types/chain.rs:308]: are these compatible with etherscan API?

Comment (AmeanAsad) [reply ethers-core/src/types/chain.rs:308]: They are not

**Comment by prestwich** [comment]
we can't guarantee that ethers will behave properly for any virtualized VM 🤔

**Comment by AmeanAsad** [comment]
@prestwich the filecoin virtual machine is EVM compatible (idk if that helps). The main reason for this request is to enable gas estimation in `Foundry` which uses ethers `Chain` enum. As of now that is the only functionality required from this change. Is that acceptable?

**Comment by prestwich** [comment]
the FVM has a virtualized EVM, this is not necessarily the same thing as being EVM-compatible 😅

File: ethers-core/src/types/chain.rs (hunk 1 above)
Comment (gakonst) [reply ethers-core/src/types/chain.rs:308]: Hmm then we cannot use them. Where can we read more about the filecoin evm spec?

Comment (AmeanAsad) [reply ethers-core/src/types/chain.rs:308]: @gakonst  Here are some resources regarding docs/specs:
- [Spec](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0030.md)
- [Official docs](https://docs.filecoin.io/developers/smart-contracts/filecoin-virtual-machine/)

**Comment by DaniPopes** [comment]
I think this is ok, given that:
- it's on [Chainlist](https://chainlist.org/chain/314)
- they have docs for usage with [Metamask](https://docs.filecoin.io/developers/smart-contracts/how-tos/add-to-metamask/) and [Hardhat](https://docs.filecoin.io/developers/smart-contracts/hardhat/)

Although they have their own precompiles

**Review by gakonst — APPROVED** [review]
K - given no block explorer handlign im fine w this

**Pull request merged**